### PR TITLE
Feature/259 generic trainer test

### PR DESCRIPTION
**Pull request by AndSt**

### CODE REVIEW

File: tests/trainer/test_auto_trainer.py [hunk 1]
```diff
@@ -1,43 +1,27 @@
...
-    metrics = trainer.test(model_input_x, y_labels)
-
-    # Check whether the code ran up to here
-    assert 2 == 2
+def test_auto_train(std_trainer_input_1):
+    (
+        model,
+        model_input_x, rule_matches_z, mapping_rules_labels_t,
+        y_labels
+    ) = std_trainer_input_1
```
Comment (marina-sp) [inline tests/trainer/test_auto_trainer.py:11]: Do we need a separate input for this, could not we use std_trainer_input_2 for this too?

File: knodle/trainer/snorkel/snorkel.py [hunk 2]
```diff
@@ -38,13 +38,18 @@ def _snorkel_denoising(self, model_input_x, rule_matches_z):
...
+        fitting_kwargs = {}
+        if self.trainer_config.seed is not None:
+            fitting_kwargs["seed"] = self.trainer_config.seed
+
         label_model.fit(
             L_train,
             n_epochs=self.trainer_config.label_model_num_epochs,
             log_freq=self.trainer_config.label_model_log_freq,
-            seed=self.trainer_config.seed
+            **fitting_kwargs
```
[the remark this answers is not in the document]
Comment (AndSt) [reply knodle/trainer/snorkel/snorkel.py:49]: Because you can not set seed=None. It's either a number or not there ..

File: tests/trainer/test_auto_trainer.py (hunk 1 above)
Comment (AndSt) [reply tests/trainer/test_auto_trainer.py:11]: We already created 2 scenarios, so I thought let's keep it like it is, as it's only more thorough testing. But if you think, it'd be more consistent I could use 1 only :)

File: knodle/trainer/snorkel/snorkel.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (marina-sp) [reply knodle/trainer/snorkel/snorkel.py:49]: Yes i actually encountered this issue in the meantime and had to do exactly the same as you did here :D
